scripts/dpdk_setup_ports.py

```diff
@@ -1031,30 +1043,45 @@ def split_pci_key(self, pci_id):
         return pci_id.split('/')[0]
 
     def _get_cpu_topology(self):
-        cpu_topology_file = '/proc/cpuinfo'
-        # physical processor -> physical core -> logical processing units (threads)
         cpu_topology = OrderedDict()
-        if not os.path.exists(cpu_topology_file):
-            raise DpdkSetup('File with CPU topology (%s) does not exist.' % cpu_topology_file)
-        with open(cpu_topology_file) as f:
-            for lcore in f.read().split('\n\n'):
-                if not lcore:
-                    continue
-                lcore_dict = OrderedDict()
-                for line in lcore.split('\n'):
-                    key, val = line.split(':', 1)
-                    lcore_dict[key.strip()] = val.strip()
-                if 'processor' not in lcore_dict:
-                    continue
-                numa = int(lcore_dict.get('physical id', -1))
-                if numa not in cpu_topology:
-                    cpu_topology[numa] = OrderedDict()
-                core = int(lcore_dict.get('core id', lcore_dict['processor']))
-                if core not in cpu_topology[numa]:
-                    cpu_topology[numa][core] = []
-                cpu_topology[numa][core].append(int(lcore_dict['processor']))
+
+        # Find the total number of CPUs (logical cores)
+        base_path = "/sys/devices/system/cpu"
+        fd = open("{}/kernel_max".format(base_path))
```

Comment:
I would open file with "with open() ..." and remove "close"

Reply:
Done.